### ISSUE TITLE
~/data issue - my participant demo user doesn't see all files, permissions acting up

### ISSUE THREAD

**Issue by consideRatio** [neurohackademy/nh2020-jupyterhub#99]
As my normal user with read/write rights and ~/data mounted as a NFS drive, I see the following.

![image](https://user-images.githubusercontent.com/3837114/88641747-e787f780-d0bf-11ea-8c11-d25bd35a2937.png)

As my participant trial user, with read only rights and ~/data mounted as a hostPath volume with a rsync-ed cache of the NFS drive, I see the following.

![image](https://user-images.githubusercontent.com/3837114/88642026-3df53600-d0c0-11ea-91ba-ca22838a46e3.png)

And before, I could only see the folders up to sub-04, and as can be noted in the permissions, it seems like there is some differences between sub-04 and sub-05 and onwards.

---

I found the following when inspecting k8s events of my nodes.

```
  Warning  OOMKilling  53s (x2282 over 24h)  kernel-monitor, gke-nh-2020-users-ultramem-40-e3dc06f2-p7m4  (combined from similar events): Memory cgroup out of memory: Kill process 1687293 (rsync) score 995 or sacrifice child
Killed process 1687295 (rsync) total-vm:14072kB, anon-rss:548kB, file-rss:592kB, shmem-rss:0kB
```

```
$ kubectl logs nfs-node-cacher-8vvhf 
sending incremental file list
ds005-preproc/fmriprep/sub-01/anat/
ds005-preproc/fmriprep/sub-01/anat/sub-01_hemi-L_inflated.surf.gii
Killed
        819.20K  26%  728.65kB/s    0:00:03  sending incremental file list
ds005-preproc/fmriprep/sub-01/anat/
ds005-preproc/fmriprep/sub-01/anat/sub-01_hemi-L_inflated.surf.gii
rsync: [receiver] write error: Broken pipe (32)
rsync error: error in socket IO (code 10) at io.c(820) [receiver=3.1.3]

rsync: [sender] write error: Broken pipe (32)
rsync error: error in socket IO (code 10) at io.c(820) [sender=3.1.3]
         32.77K   1%    0.00kB/s    0:00:00  sending incremental file list
rsync: [receiver] write error: Broken pipe (32)
rsync error: error in socket IO (code 10) at io.c(820) [receiver=3.1.3]
Killed
```

My k8s resource requests for these pods have been...

```yaml
          resources:
            requests:
              cpu: "0.5"
              memory: 4Gi
            limits:
              cpu: "1"
              memory: 4Gi
```

The status of the pods are...

![image](https://user-images.githubusercontent.com/3837114/88643030-79dccb00-d0c1-11ea-826b-53009352f551.png)

And their resource usage with `kubectl top pods` are...

![image](https://user-images.githubusercontent.com/3837114/88643229-bb6d7600-d0c1-11ea-8739-5001301e0eb3.png)

I think what has happened is that the memory requirements of copying all files at once was too large and failed, but the older nodes that got to copy a smaller set of files incrementally as the NFS folder grew managed to do it without running out of memory.

The amount of data currently in the NFS ~/data directory is...

![image](https://user-images.githubusercontent.com/3837114/88643512-130be180-d0c2-11ea-89a4-3f7c49dbccf2.png)

**Comment by consideRatio** [comment]
#100 resolved the issue, and the pods now report the following memory usage with `kubectl top pods`.

![image](https://user-images.githubusercontent.com/3837114/88644957-df31bb80-d0c3-11ea-8269-c9bd0ed9c623.png)

While this was a quick fix, it's not very nice. What if the folder grows even larger? Some googling made me [find the suggestion of using `--no-inc-recursive`](https://www.lowendtalk.com/discussion/6734/rsync-out-of-memory-try-this).

Trying it out, the new pods got the following memory usage (they didn't need to copy anything though as it was already up to date).

![image](https://user-images.githubusercontent.com/3837114/88646982-df32bb00-d0c5-11ea-8dd7-d0d39770c1bb.png)

When trying it out on a new node without anything rsynced to it already, the node reported the following.

![image](https://user-images.githubusercontent.com/3837114/88649272-04c0c400-d0c8-11ea-8a25-5c8997c9d267.png)

---

In other words, `--no-inc-recursive` didn't do the trick it seems.
